### PR TITLE
Gb/upload media file

### PR DESCRIPTION
**Pull request by marecar3**
**Description :**
This PR introduces image file upload from Gutenberg which implementation is based on this [issue.](https://github.com/wordpress-mobile/gutenberg-mobile/issues/206).

This is the first and base iteration of a task that should be implemented. Initial idea is to present an image picker from gutenberg editor which will trigger the upload process when the user picks an image.

**Following PR's :**
[gutenberg-mobile 
](https://github.com/wordpress-mobile/gutenberg-mobile/pull/458)[gutenberg](https://github.com/WordPress/gutenberg/pull/13128)

**Note :** 
- Missing final colors and dimensions for progress bar as it's hard to take it from the png.
- Missing the error state (when an upload fails)
- Not sure if edit toolbar button should open media picker or image chooser after an image is uploaded with success. 

@iamthomasbishop please can you help me with the above notes? Thanks!

ezgif com-video-to-gif 1

### CODE REVIEW

**Comment by iamthomasbishop** [comment]
Hey @marecar3  looking good!

Here are the specs, which I'll also drop in the Zeplin project asap.

> final colors and dimensions for progress bar

* Progress bar: #0087BE (`$blue-wordpress`)
* Track: #c8d7e1 (`$gray-lighten-20`)
* Height: 5pt
* Width: 100% of image

> error state (when an upload fails)

A visual example of the `loading` and `error` states:

screen shot 2018-12-28 at 2 19 12 pm

* Overlay: #2E4453 (`$gray-dark`) at 60% opacity – this should match Aztec, so we need to double check it matches. // cc @etoledom is this correct?
* Icon and text are both #FFFFFF
* We're using `gridicon-refresh` for the icon, but we can use any available icon that has the same effect
* Text is 14pt SF Pro Text Semibold on iOS, and I'm thinking it's 14sp Roboto in Medium

> Not sure if edit toolbar button should open media picker or image chooser after an image is uploaded with success.

I would match what exists on Gutenberg web here (although I do think the pencil icon here is confusing). So tapping the pencil on the toolbar would open the same picker that you chose the current image from. I'm not a huge fan of this workflow, but for consistency and brevity, let's start there.

Another option would be to utilize the same flow that we have built for Aztec for now, which looks (roughly) like this:

screen shot 2018-12-28 at 3 09 55 pm

Inside the wider Gutenberg flow, it could look like this:

gb alpha - insert image flow - ios i1

I've got other suggestions, but for the alpha/beta let's keep it simple.

**Comment by hypest** [comment]
Hey @marecar3 , not sure if it was part of the goal for this PR but, it seems that leaving the editor and re-entering while the image is still uploading doesn't seem to work correctly:
1. The block validation fails upon re-entering
2. The image upload indicator doesn't seem to update on re-enter and it never gets cleared

Do you think we can address those issues during this PR?

**Comment by mzorz** [comment]
> it seems that leaving the editor and re-entering while the image is still uploading doesn't seem to work correctly

hi @hypest 👋 This I think exceeds the scope of this particular PR, I've tracked it here https://github.com/wordpress-mobile/gutenberg-mobile/issues/459 and I was planning to tackle it (in conjunction with Marko when he's back) as I intend to do this wiring in the new `GutenbergEditPostActivity` introduced in https://github.com/wordpress-mobile/WordPress-Android/pull/8874

For some context, keeping states up to date for in-progress uploads was one of the trickiest part of the async media upload project before (see https://github.com/wordpress-mobile/WordPress-Android/pull/5780 for one example of various cases that have to be taken into account), so I'd like to bring that "case-awareness" here as well

**Comment by mzorz** [comment]
Clarification:
> The block validation fails upon re-entering

This is something we should still investigate and work on as needed. I was talking only about media progress re-attachment in my previous comment

**Comment by hypest** [comment]
> I think we may want to change the base branch to a feature branch in WPAndroid repo before actually merging though. Leaving that for @hypest to decide.

Thanks for the ping @mzorz . I think that wpandroid's develop needs to be treated as a branch more stable than the gutenberg-mobile's develop so yeah, let's target a feature branch for the media upload work.

By the way, I re-triggered CircleCI.

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java [hunk 1]
```diff
@@ -111,6 +112,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
                     @Override public void onMediaLibraryButtonClick() {
                         onToolbarMediaButtonClicked();
                     }
+
+                    @Override
+                    public void onUploadMediaButtonClick() {
```
[the remark this answers is not in the document]
Comment (marecar3) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: @mzorz thanke for the update, fixed.

**Comment by marecar3** [comment]
Base branch is changed to `origin/gb/feature_upload_media_file` .

**Pull request merged**